app/models/bulk_upload_data_set.rb

```diff
@@ -1242,8 +1242,12 @@ def insert_data
           t.site_id = row['site_id']
 
           # Modify each Trait class instance so that date strings are
-          # interpreted as being in the time zone of the trait site
-          # (or UTC, if the trait site time_zone column is null)
+          # interpreted as being in the time zone of the trait site (or UTC, if
+          # the trait site time_zone column is null).  Note that in the Bulk
+          # Upload Wizard, the validation step will prevent reaching this code
+          # if a site without time zone is in the data file.  But nothing
+          # prevents a site without time zone from being specified
+          # interactively.
```

Comment:
@gsrohde you mean time zones aren't assigned automatically based on location?

Reply:
@dlebauer That is correct, unfortunately.  When I added the `time_zone` column, I did create and use table of time-zone/location data to facilitate automatically populating the `time_zone` column for existing sites, but there is nothing built-in to do this when adding a new site.